lib/util/http/github.ts

```diff
@@ -315,9 +320,18 @@ export class GithubHttp extends Http<GithubHttpOptions> {
           if (!process.env.RENOVATE_PAGINATE_ALL && opts.paginate !== 'all') {
             lastPage = Math.min(pageLimit, lastPage);
           }
+          const baseUrl = opts.baseUrl;
           const queue = [...range(2, lastPage)].map(
             (pageNumber) => (): Promise<HttpResponse<T>> => {
-              const nextUrl = new URL(linkHeader.next.url, opts.baseUrl);
+              const parsedUrl = new URL(linkHeader.next.url, baseUrl);
+              const rebasePaginationLinks =
```

Comment:
move this outside the loop, so we don't have to recompute on every iteration.

Reply:
moved https://github.com/renovatebot/renovate/pull/19888/commits/9af856b9a4828b595b940ec3ad4f23229cf262bf